sycl/include/CL/sycl/multi_ptr.hpp

```diff
@@ -70,9 +77,21 @@ template <typename ElementType, access::address_space Space> class multi_ptr {
     return *this;
   }
 
+#if defined(RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR)
+  using ReturnPtr =
+      typename std::conditional<Space == access::address_space::constant_space,
+                                const ElementType *, ElementType *>::type;
+  using ReturnRef =
```

Comment:
```suggestion
  using ReturnRef = ReturnConstRef
```

Reply:
I've made something like this...

Reply:
Something more C++-conformant, dare I say :)